recipes/pybind11/all/conanfile.py

```diff
@@ -38,6 +38,8 @@ def package(self):
         cmake.install()
         os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Config.cmake"))
         os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11ConfigVersion.cmake"))
+        pybind11tools_path = os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Tools.cmake")
+        tools.replace_in_file(pybind11tools_path, "PYBIND11_INCLUDE_DIR", "pybind11_INCLUDE_DIRS")
```

Comment:
@jgsogo maybe this is not a good solution when consuming this package from the cmake_package_multi generator?

Reply:
What I see here is that inside the file `pybind11Tools` we have a problem. We shouldn't use `pybind11_INCLUDE_DIRS` but the `pybind11::pybind11` target... but it won't work if using just a `cmake` generator.

I only have questions here: can we use the `CONAN_LIB::...` target? Probably not the same name for `cmake_find_package` generators. Should we create always the target `pybind11::pybind11` even for `cmake` generator? What about old CMake without targets supports (do we care about so old versions for this recipe?)? 

A different way to go is add a list to different include directories (it already has a bunch of them). We can add the generator expression that works for the `cmake_find_package[_multi]` generator:
```cmake
  target_include_directories(${target_name} ${inc_isystem}
    PRIVATE ${pybind11_INCLUDE_DIRS}  # from project CMakeLists.txt
    PRIVATE ${pybind11_INCLUDE_DIR}  # from pybind11Config
    PRIVATE ${PYTHON_INCLUDE_DIRS}
)
```

Here we have a similar problem as the one be have in the `protobuf` package. We need to hardcode in a file something that works for any generator that we will define afterward. Can we workaround it? Is this one reason why we need a proper `deploy` function with the additional install folder in the cache?

I know this is not an answer, only questions, but I need to think more about this issue.